$OpenBSD: patch-camel_camel-db_c,v 1.1 2009/02/17 10:56:27 ajacoutot Exp $

Fix  64-bit time_t problem which can lead to mail database corruption.

--- camel/camel-db.c.orig	Mon Feb  2 04:19:08 2009
+++ camel/camel-db.c	Tue Feb 17 11:11:12 2009
@@ -1213,11 +1213,11 @@ write_mir (CamelDB *cdb, const char *folder_name, Came
 
 	/* NB: UGLIEST Hack. We can't modify the schema now. We are using msg_security (an unsed one to notify of FLAGGED/Dirty infos */
 
-	ins_query = sqlite3_mprintf ("INSERT OR REPLACE INTO %Q VALUES (%Q, %d, %d, %d, %d, %d, %d, %d, %d, %d, %d, %ld, %ld, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q )", 
+	ins_query = sqlite3_mprintf ("INSERT OR REPLACE INTO %Q VALUES (%Q, %d, %d, %d, %d, %d, %d, %d, %d, %d, %d, %lld, %lld, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q, %Q )", 
 			folder_name, record->uid, record->flags,
 			record->msg_type, record->read, record->deleted, record->replied,
 			record->important, record->junk, record->attachment, record->dirty,
-			record->size, record->dsent, record->dreceived,
+			record->size, (long long)record->dsent, (long long)record->dreceived,
 			record->subject, record->from, record->to,
 			record->cc, record->mlist, record->followup_flag,
 			record->followup_completed_on, record->followup_due_by, 
